test/temporal.jl

```diff
@@ -0,0 +1,173 @@
+
+        @testset "inds" begin
+            @test Transforms.apply(x, hod; inds=2:5) == expected[2:5]
+            @test Transforms.apply(x, hod; dims=:) == expected
+            @test Transforms.apply(x, hod; dims=1) == expected
+            @test Transforms.apply(x, hod; dims=1, inds=[2, 3, 4, 5]) == expected[2:5]
+        end
+
+        @testset "DST" begin
+            x = ZonedDateTime(2020, 3, 7, 9, 0, tz"America/New_York"):Hour(1):ZonedDateTime(2020, 3, 8, 9, 0, tz"America/New_York")
+
+            # expected result skips the DST transition hour of 2
+            expected_dst = [9:23..., 0, 1, 3:9...]
+
+            @test Transforms.apply(x, hod) == expected_dst
+            @test hod(x) == expected_dst
+        end
+    end
+
+    @testset "Matrix" begin
+        x = [
+            DateTime(2020, 1, 1, 1, 0):Hour(1):DateTime(2020, 1, 1, 3, 0);
+            DateTime(2020, 1, 1, 9, 0):Hour(1):DateTime(2020, 1, 1, 11, 0)
+        ]
+        M = reshape(x, 3, 2)
+        expected = [1 9; 2 10; 3 11]
+
+        @test Transforms.apply(M, hod) == expected
+        @test hod(M) == expected
+
+        # Test the tranform was not mutating
+        @test M != expected
+
+        @testset "dims = $d" for d in (Colon(), 1, 2)
+            @test Transforms.apply(M, hod; dims=d) == expected
+            @test hod(M; dims=d) == expected
+        end
+
+        @testset "inds" begin
+            @test Transforms.apply(M, hod; inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(M, hod; dims=:, inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(M, hod; dims=1, inds=[2]) == reshape(expected[[2, 5]], 1, 2)
+            @test Transforms.apply(M, hod; dims=2, inds=[2]) == reshape(expected[[4, 5, 6]], 3, 1)
+        end
+    end
+
+    @testset "AxisArray" begin
+        x = [
+            DateTime(2020, 1, 1, 1, 0):Hour(1):DateTime(2020, 1, 1, 2, 0);
+            DateTime(2020, 1, 1, 9, 0):Hour(1):DateTime(2020, 1, 1, 10, 0);
+            DateTime(2020, 1, 1, 10, 0):Hour(1):DateTime(2020, 1, 1, 11, 0)
+        ]
+        M = reshape(x, 2, 3)
+        A = AxisArray(M, foo=["a", "b"], bar=["x", "y", "z"])
+        expected = [1 9 10; 2 10 11]
+
+        @test Transforms.apply(A, hod) == expected
+        @test hod(A) == expected
+
+        @testset "dims = $d" for d in (Colon(), 1, 2)
+            transformed = Transforms.apply(A, hod; dims=d)
+            # AxisArray doesn't preserve the type it operates on
+            @test transformed isa AbstractArray
+            @test transformed == expected
+        end
+
+        @testset "inds" begin
+            @test Transforms.apply(A, hod; inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(A, hod; dims=:, inds=[2, 3]) == expected[[2, 3]]
+            @test Transforms.apply(A, hod; dims=1, inds=[2]) == reshape(expected[[2, 4, 6]], 1, 3)
+            @test Transforms.apply(A, hod; dims=2, inds=[2]) == reshape(expected[[3, 5]], 2, 1)
```

Comment:
```suggestion
            @test Transforms.apply(A, hod; dims=2, inds=[2]) == reshape(expected[[3, 4]], 2, 1)
```
If I understand correctly, `dims=2, inds=[2]` is selecting the middle column, which is global index 3 and 4. So I think this passed by coincidence of global index 4 and 5 having the same value?

Reply:
I think in general I'd prefer to write the expected result more explicitly (where it's not trivial) rather rely on `reshape` and indexing.

Instead of `reshape(expected[[3, 4]], 2, 1)` say `[9, 10]` 

It's more clear what the expected answer is.